tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -135,6 +135,28 @@ def last_commit(self, tp):
         else:
             return None
 
+# This needs to be used for cooperative and consumer protocol
+class IncrementalAssignmentConsumerEventHandler(ConsumerEventHandler):
```

Comment:
Why can we not implement this in the normal ConsumerEventHandler? It seems it's making incorrect assumptions about the consumer rebalance listener (since previously owned partitions are not guaranteed to be reported in onassinged)

Reply:
I believe the current listener assumes Eager protocol so it is not making the incorrect assumptions.  This (incremental handler) would probably work for Eager as well but I thought it would be clearer to distinguish the two.

Reply:
Seems to me that it would have been cleaner to have a single implementation that treats the value that is passed in `onAssigned` correctly (it is always incremental by contract, just happens to be non-incremental in the eager case), instead of having two implementations now. But I'll leave it to a follow-up PR to clean it up, if you agree, and merge this now.